### PR TITLE
fix(zone.js): Promise.resolve(subPromise) should return subPromise

### LINKED ISSUE #53380: Angular Zones break promises in `openai` package (return null)
### Which @angular/* package(s) are the source of the bug?

zone.js

### Is this a regression?

No

### Description

[Users making requests](https://github.com/openai/openai-node/issues/232#issuecomment-1820088056) with the `openai` package while using Zones receive `null` on all requests.

Eg; 

```ts
const response = await this.openai!.chat.completions.create({
  model: 'gpt-3.5-turbo',
  messages: [
    {role: "system", content: "Explain me the purpose of life"},
  ],
});
// response === null, but shouldn't be
```

[Users report](https://github.com/openai/openai-node/issues/232#issuecomment-1806001114) that upgrading zone.js to ^0.14.2 allows things to work with `.then()`, but not with `await`. 

This is likely related to the `openai` package [subclassing Promise](https://github.com/openai/openai-node/blob/a048174c0e53269a01993a573a10f96c4c9ec79e/src/core.ts#L79) to add helper methods, [like `.asResponse()`](https://github.com/openai/openai-node/blob/a048174c0e53269a01993a573a10f96c4c9ec79e/src/core.ts#L79).

### Please provide a link to a minimal reproduction of the bug

https://github.com/openai/openai-node/issues/232#issuecomment-1820088056

### Please provide the exception or error you saw

```true
completion is `null`
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
I am a maintainer of `openai`, not an Angular user. Users have reported this with Angular v17 (and earlier) and zone.js@0.14.2 (and earlier).
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JiaLiPassion**
Close https://github.com/angular/angular/issues/53380

In the original `Promise` impelmentation, zone.js follow the spec from
https://promisesaplus.com/#point-51.

```
const p1 = Promise.resolve(1);
const p2 = Promise.resolve(p1);

p1 === p2; // false
```
in this case, `p2` should be the same status with `p1` but they are
still different instances.

And for some edge case.

```
class MyPromise extends Promise {
  constructor(sub) {
    super((res) => res(null));
    this.sub = sub;
  }
  then(onFufilled, onRejected) {
    this.sub.then(onFufilled, onRejected);
  }
}

const p1 = new Promise(setTimeout(res), 100);
const myP = new MyPromise(p1);
const r = await myP;
r === 1; // false
```

So in the above code, `myP` is not the same instance with `p1`,
and since `myP` is resolved in constructor, so `await myP` will
just pass without waiting for `p1`.

And in the current `tc39` spec here https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-promise-resolve
`Promise.resolve(subP)` should return `subP`.

```
const p1 = Promise.resolve(1);
const p2 = Promise.resolve(p1);

p1 === p2; // true
```

So the above `MyPromise` can wait for the `p1` correctly.

### CODE REVIEW

**Review by alan-agius4 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -234,7 +234,8 @@ jobs:
         # Install
       - run: yarn --cwd packages/zone.js install --frozen-lockfile --non-interactive
         # Run zone.js tools tests
-      - run: yarn --cwd packages/zone.js promisetest
+      # Skip promisetest since the test cases has conflicts with tc39 spec.
+      # - run: yarn --cwd packages/zone.js promisetest
```
[the remark this answers is not in the document]
Comment (JiaLiPassion) [reply .github/workflows/ci.yml]: @devversion, yeah, I just removed this test. thanks.

[the remark this answers is not in the document]
Comment (devversion) [reply .github/workflows/ci.yml]: Thanks Jia! can we remove the `package.json` entry and related deps as well?

**Comment by alan-agius4** [comment]
[TGP](https://test.corp.google.com/OCL:589092156:BASE:589091977:1702039957285:1580adfe)

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (JiaLiPassion) [reply .github/workflows/ci.yml]: @devversion , sure, I removed the packge.json entry/deps and README.md doc. Thanks.

**Review by devversion — APPROVED** [review]

**Pull request closed without merging**